lib/common/puyo/stepper.js

```diff
@@ -264,10 +268,19 @@ class DuelStepper extends BasicStepper {
     const hasStepped = {};
     const nuisanceReceivers = [];
     const eventsForPlayer = this.distributeEvents(state, events);
-
     let gameOver = false;
+    let terminated = false;
+    let terminationReason;
+    let terminatedPlayer;
+
     ++state.time;
 
+    if (events.find(event => event.type === 'terminated')) {
+      terminated = true;
+      terminationReason = event.reason;
```

Comment:
While this might work because of the arrow function syntax that messes up with the scope, it's still kinda like abusing another badly designed feature in JavaScript. Not sure whether I like it.